File: Baekjoon/Class2/35.18110 solved.ac.py
import sys
input = sys.stdin.readline

def positive_round(num: float):
    return int(num + 0.5)

def main():
    n = int(input())
    if n == 0:
        print(0)
        sys.exit(0)
    opinions = [int(input()) for _ in range(n)]
    opinions.sort() #정렬
    exclusion = positive_round(n * 0.15) #절사 범위
    total = sum([opinions[i] for i in range(exclusion, n - exclusion)]) #절사
    print(positive_round(total / (n - 2 * exclusion))) # 평균

main()

File: Baekjoon/Class2/35.18110 solved.ac.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n").map(Number);

const n = input[0];
if(n === 0) {
    console.log(0);
    process.exit();
}

const opinions = input.slice(1).sort((a, b) => a - b); // 정렬
let total = 0;
const exclusion = Math.round(n * 0.15); // 절사 범위
for(let i = exclusion; i < n - exclusion; i++) // 절사
    total += opinions[i];
console.log(Math.round(total / (n - 2 * exclusion))); // 평균